test/func/mocha.opts

```diff
@@ -1,3 +1,3 @@
 --require test/func/setup.js
 --recursive
---timeout 10000
+--timeout 20000
```

Comment:
I observed this integrating the functional testing suite in another project as well. The 10000 timeout limit didn't cut it anymore. I wasn't sure if the right thing was to try to narrow it down to the particular cases that were pushing it over, or just to set it higher in general.

In principle, setting a high timeout limit at the beginning feels more reasonable to me than sprinkling in overrides in tests that happen to be slow, so I went with this for now.